### DISCUSSION TITLE
Enhancing the method used to get the user input

### DISCUSSION THREAD

**Discussion by woosaaahh** [woosaaahh/sj.nvim#11] [Ideas]
The discussion is based on this comment : https://github.com/woosaaahh/sj.nvim/issues/10#issuecomment-1264880881 (thanks @jemag !)

The very fist implementation was using a user command with inccommand and preview but the "interactivity" wasn't that great.

I like the idea of using a buffer to let the users type as they want.
I tried a bit prompt-buffers and even if they provide some nice features, I'm not comfortable with them...

So I made a custom search input with a basic floating window and a buffer.
I'll have more code to write to handle the buffer but it'll give me more flexibility.

Here is what I have so far : 
![sj-search_box](https://user-images.githubusercontent.com/111681540/193676462-719ebffb-63e5-40e9-9b24-0f45d613b394.gif)

The search box will appear top/right of the current window.
You can see a prompt on the left with a little magnifying glass and a red text on the right.
The red text is just the length of user input, not the matches count yet.
This is just a little test to try some live update while typing.

Until the end of the week, I'll have less time to be online and to work on the plugin but I'll give updates here when needed.

**Comment by jemag** [comment]
I think the input box looks great. Although, I think I might, personally,  prefer having the location being globally consistent  just to avoid the small mental overhead of finding it(for example, right now it is always in the same place, similar to regular vim search).

**Reply by woosaaahh** [reply]
You will be able to set custom config/location for the window.

With the current prototype, you can also load the buffer in any existing window.
The goal behind this a to have kind of a search buffer with history like the command line window (`q:` in normal mode).
You'll be able to edit previous search patterns and run a search for the pattern on cursor line (`Enter`).

**Comment by woosaaahh** [comment]
I made huge progress on the new input system but sadly, it is not done yet and there is still a lot to do !

**Comment by woosaaahh** [comment]
Btw, `0.5` has been merged to `main`.
